pandas/io/packers.py

```diff
@@ -255,17 +261,36 @@ def unconvert(values, dtype, compress=None):
     if not as_is_ext:
         values = values.encode('latin1')
 
-    if compress == u'zlib':
-        import zlib
-        values = zlib.decompress(values)
-        return np.frombuffer(values, dtype=dtype)
-
-    elif compress == u'blosc':
-        import blosc
-        values = blosc.decompress(values)
-        return np.frombuffer(values, dtype=dtype)
+    if compress:
+        if compress == u'zlib':
+            from zlib import decompress
+        elif compress == u'blosc':
+            from blosc import decompress
+        else:
+            raise ValueError("compress must be one of 'zlib' or 'blosc'")
+
+        values = decompress(values)
+        if sys.getrefcount(values) == 2:
```

Comment:
I actually thing the ideal fix would be for blosc and zlib to have unpack functions that give back bytearrays so that we can do this in a more safe way. I will investigate if this is already available.